pkg/cloudprovider/providers/photon/photon.go

```diff
@@ -286,65 +301,122 @@ func (pc *PCCloud) List(filter string) ([]k8stypes.NodeName, error) {
 
 // NodeAddresses is an implementation of Instances.NodeAddresses.
```

Comment:
This function is part of the Instance interface's API function. The definition is created by kubernetes and this is the only one line they added for it... This is more like a template thing.